src/dill/gauge-proxies/sidechain-gauge-proxy.sol

```diff
@@ -0,0 +1,172 @@
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    IERC20Upgradeable public PICKLE;
+
+    address[] internal _tokens;
+    address bridgeClient;
+
+    enum GaugeType {
+        REGULAR,
+        VIRTUAL
+    }
+
+    struct Gauge {
+        GaugeType gaugeType;
+        address gaugeAddress;
+    }
+
+    mapping(address => Gauge) public gauges;
+    mapping(uint256 => uint256) public periodRewardAmount;
+    mapping(uint256 => int256[]) public periodGaugeWeights;
+
+    struct periodData {
+        uint256 amount;
+        int256[] weights;
+    }
+    mapping(uint256 => periodData) public periods; // periodID => periodData
+    mapping(uint256 => bool) public distributedForPeriod;
+    uint256 public periodToDistribute;
+
+    function tokens() external view returns (address[] memory) {
+        return _tokens;
+    }
+
+    function getGauge(address _token) external view returns (Gauge memory) {
+        return gauges[_token];
+    }
+
+    function initialize(
+        address pickleToken,
+        address _bridgeClient,
+        uint256 _periodToDistribute
+    ) public initializer {
+        periodToDistribute = _periodToDistribute;
+        governance = msg.sender;
+        PICKLE = IERC20Upgradeable(pickleToken);
+        bridgeClient = _bridgeClient;
+    }
+
+    // Add new token gauge
+    function addGauge(address _token, address _gaugeAddress) external {
+        require(msg.sender == governance, "!gov");
+        require(gauges[_token].gaugeAddress == address(0x0), "exists");
+
+        gauges[_token].gaugeAddress = _gaugeAddress;
+        gauges[_token].gaugeType = GaugeType.REGULAR;
+
+        _tokens.push(_token);
+    }
+
+    // Add new token virtual gauge
+    function addVirtualGauge(
+        address _token,
+        address _jar,
+        address _gaugeAddress
+    ) external {
+        require(msg.sender == governance, "!gov");
+
+        require(gauges[_token].gaugeAddress == address(0x0), "exists");
+
+        gauges[_token].gaugeAddress = _gaugeAddress;
+        gauges[_token].gaugeType = GaugeType.VIRTUAL;
+
+        _tokens.push(_token);
+    }
+
+    function distribute(uint256 _start, uint256 _end) external {
+        require(_start < _end, "SidechainGaugeProxy: bad _start");
+        require(_end <= _tokens.length, "SidechainGaugeProxy: bad _end");
+        require(
+            msg.sender == governance,
+            "GaugeProxyV2: only governance can distribute"
+        );
+        uint256 _periodToDistribute = periodToDistribute;
+        require(
+            distributedForPeriod[_periodToDistribute] == false,
+            "Already distributed for given period"
+        );
+        require(
+            periods[_periodToDistribute].weights.length == 0 &&
+                periods[_periodToDistribute].amount == 0,
+            "All period distribution compleated"
+        );
+        int256[] memory _weights = periods[periodToDistribute].weights;
```

Comment:
If the `sendRewards` and `distribute` calls might ever be de-coupled, would we want to use the previous week's weights if the current week's `sendRewards` call was missed (similar to that of mainnet's GaugeProxyV2)?